src/noit_filters.c

```diff
@@ -655,6 +685,76 @@ noit_apply_filterrule_metric(filterrule_t *r,
   return mtev_true;
 }
 
+static int
+noit_add_measurement_tag(filterrule_t *r,
+                         metric_t *metric,
+                         char **expanded_metric_name,
+                         noit_metric_tagset_t *mtset) {
+  if (!r || !metric || !expanded_metric_name || !mtset || !r->measurement_tag.add_measurement_tag_cat || mtset->tag_count >= MAX_TAGS) {
+    return -1;
+  }
+  char encoded_nametag[NOIT_TAG_MAX_PAIR_LEN+1];
+  char decoded_nametag[NOIT_TAG_MAX_PAIR_LEN+1];
+  const char *cat = r->measurement_tag.add_measurement_tag_cat;
+  const char *val = r->measurement_tag.add_measurement_tag_val ? r->measurement_tag.add_measurement_tag_val : "";
+  snprintf(decoded_nametag, sizeof(decoded_nametag), "%s%c%s", cat, NOIT_TAG_DECODED_SEPARATOR, val);
```

Comment:
what if there is a cat but empty val?  does this still work?

Reply:
Cat with empty val is valid, val with empty cat is not